src/requests.js

```diff
@@ -20,28 +21,78 @@ function safeHeader(safe, last_modified) {
   return {"If-None-Match": "*"};
 }
 
-/**
- * @private
- */
-export function createBucket(bucketName, options={}) {
-  if (!bucketName) {
-    throw new Error("A bucket name is required.");
-  }
-  // Note that we simply ignore any "bucket" option passed here, as the one
-  // we're interested in is the one provided as a required argument.
-  const { data={}, headers, permissions, safe } = {
+
+function _create(path, {data, permissions}, options={}) {
+  const { headers, safe } = {
     ...requestDefaults,
     ...options,
   };
-  return {
-    method: "PUT",
-    path: endpoint("bucket", bucketName),
+  return removeUndefined({
+    method: data && data.id ? "PUT" : "POST",
+    path,
     headers: {...headers, ...safeHeader(safe)},
     body: {
       data,
       permissions
     }
+  });
+}
+
+function _update(path, {data, permissions}, options={}) {
+  const {
+    headers,
+    safe,
+    patch,
+  } = {...requestDefaults, ...options};
+  const { last_modified } = { ...data, ...options };
+
+  if (Object.keys(omit(data, "id", "last_modified")).length === 0) {
+    data = undefined;
+  }
+
+  return removeUndefined({
```

Comment:
I'm surprised this is needed as `undefined` properties are removed by serialization while it's already performed when sending the request body through `execute`: https://github.com/Kinto/kinto-client/blob/master/src/base.js#L366-L369

Am I missing something here? What breaks when this call is removed?